File: Easy/1342_Number_of_Steps_to_Reduce_a_Number_to_Zero/Solution.py
from typing import Optional, List


class Solution:
    def numberOfSteps(self, num: int) -> int:
        steps = 0

        while num > 0:
            if num % 2 == 0:
                num //= 2
            else:
                num -= 1
            steps += 1
        
        return steps
        


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    print(solution.numberOfSteps(14))

    # test cases 2
    print(solution.numberOfSteps(8))

    # test cases 3
    print(solution.numberOfSteps(123))

File: Easy/1342_Number_of_Steps_to_Reduce_a_Number_to_Zero/Solution.java



class Solution {
    public int numberOfSteps(int num) {
        int steps = 0;

        while (num > 0) {
            if (num % 2 == 0) {
                num /= 2;
            } else {
                num -= 1;
            }
            steps++;
        }

        return steps;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        System.out.println(solution.numberOfSteps(14));

        // test cases 2
        System.out.println(solution.numberOfSteps(8));

        // test cases 3
        System.out.println(solution.numberOfSteps(123));
    }
}